src/app/chat/chat.service.ts

```diff
@@ -88,39 +89,49 @@ export class ChatService {
    * this method return chat list data.
    */
   getChatList(): Observable<ChatChannel[]> {
-    return this.request.get(api.getChatList).pipe(
-      map(response => {
-        if (response.success && response.data) {
-          return this._normaliseChatListResponse(response.data);
+    return this.request.chatGraphQLQuery(
+      `query getChannels {
+        channels{
+          uuid name avatar isAnnouncement isDirectMessage readonly roles unreadMessageCount lastMessage lastMessageCreated
         }
-      })
-    );
+      }`,
+      {},
+      {
+        noCache: true
+      }
+    ).pipe(map(response => {
+      if (response.data) {
+        return this._normaliseChatListResponse(response.data);
+      }
+    }));
   }
 
   /**
    * modify the Chat list response
    */
   private _normaliseChatListResponse(data): ChatChannel[] {
-    if (!Array.isArray(data)) {
+    const result = JSON.parse(JSON.stringify(data.channels));
+    if (!Array.isArray(result)) {
       this.request.apiResponseFormatError('Chat format error');
       return [];
     }
-    if (data.length === 0) {
+    if (result.length === 0) {
       return [];
     }
     const chats = [];
-    data.forEach(chat => {
+    result.forEach(chat => {
```

Comment:
Since the attributes are exactly the same, we don't need to re-assign it. 
GraphQL will make sure we get what we asked for